web/router.go

```diff
@@ -56,6 +59,55 @@ func Router(app *services.ChainlinkApplication) *gin.Engine {
 	return engine
 }
 
+// secureOptions configure security options for the secure middleware, mostly
+// for TLS redirection
+func secureOptions(config store.Config) secure.Options {
+	options := secure.Options{
+		IsDevelopment: config.Dev,
+		SSLRedirect:   config.TLSPort != 0,
+	}
+
+	if config.TLSHost != "" {
+		if config.TLSPort != 0 && config.TLSPort != 443 {
+			options.SSLHost = fmt.Sprintf("%s:%d", config.TLSHost, config.TLSPort)
+		} else {
+			options.SSLHost = config.TLSHost
+		}
+	} else if options.SSLRedirect && !options.IsDevelopment {
+		if config.TLSPort != 0 && config.TLSPort != 443 {
+			options.SSLHost = fmt.Sprintf("localhost:%d", config.TLSPort)
+		}
+	} else if !options.IsDevelopment {
+		log.Fatalf("A TLSHost must be configured when TLS redirection is enabled for a non standard port")
+	}
+
+	return options
+}
+
+// secureMiddleware adds a TLS handler and redirector, to button up security
+// for this node
+func secureMiddleware(config store.Config) gin.HandlerFunc {
+	secureMiddleware := secure.New(secureOptions(config))
+	secureFunc := func() gin.HandlerFunc {
+		return func(c *gin.Context) {
+			err := secureMiddleware.Process(c.Writer, c.Request)
+
+			// If there was an error, do not continue.
+			if err != nil {
+				c.Abort()
```

Comment:
Can we use `c.AbortWithError(err)` to bubble up the issue to the logs? Otherwise we won't know the cause.